bin/parser.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import re
+import json
+from typing import List, Dict, Any
+from collections import defaultdict
+
+WARNING_COLOR = "\033[93m"
+ENDC = "\033[0m"
+
+META_COLUMNS = [
+    "sample",
+    "cram_path",
+    "fastq_prefix",
+    "sample_supplier_name",
+    "library_type",
+    "total_reads",
+    "md5",
+]
+
+
+def parse_txt(filepath: str) -> Dict[str, Any]:
+    """
+    Parse metadata from imeta search saved in <filepath> file
+    path (dict[str]): cram file metadata from irods
+    """
+    # read file with cram metadata
+    with open(filepath, "r") as file:
+        cram_path = (
+            file.readline().strip(":\n").split(" ")[-1]
+        )  # first line includes the path of cram file
+        metadata_raw = file.read()
+
+    # Here match the following pattern in metadata:
+    ### attribute: ([\w:]+)
+    ### value: ([\w\d\S ^b]+)
+
+    pattern = re.compile(r"attribute: ([\w:]+)\nvalue: ([\w\d\S ^b]+)")
+    meta_parsed = dict(pattern.findall(metadata_raw))
+
+    # add the path of cram file to the meta
+    meta_parsed["cram_path"] = cram_path
+    return meta_parsed
+
+
+def make_fastqprefix(metadata_list: List[Dict[str, Any]]) -> None:
+    """
+    Make a name for fastq file according to the CellRanger's naming convention
+    https://www.10xgenomics.com/support/software/cell-ranger/latest/analysis/inputs/cr-specifying-fastqs
```

Comment:
10x fastq naming is based on  Illumina's `bcl2fastq` naming conventions, just to be pedantic we might as well add the extra link for completeness https://support.illumina.com/help/BaseSpace_Sequence_Hub_OLH_009008_2/Source/Informatics/BS/NamingConvention_FASTQ-files-swBS.htm